src/ddmd/func.d

```diff
@@ -5294,9 +5294,18 @@ extern (C++) final class InvariantDeclaration : FuncDeclaration
  */
 extern (C++) static Identifier unitTestId(Loc loc)
 {
+    // there's a counter per module in case there are two unit tests on one line
+    static __gshared int[const(char)*] counterPerModule;
```

Comment:
There already is/was static state. That was the source of the bug to begin with, the fact that the state reset during separate compilation. The only reason there's state in this diff is to support having more than one unittest per line, and the only reason that I considered that at all was because @yebblies told me about it.